### PR TITLE
Remove LTR component

### PR DESCRIPTION
**Pull request by diegoceccarelli**
This commit allows to use the plugin just adding the LTR query parser (no need to add a specific LTR component)

### CODE REVIEW

File: solr/contrib/ltr/example/techproducts-features.json [hunk 1]
```diff
@@ -1,26 +1,26 @@
 [
 {
     "name":  "isInStock",
-    "type": "org.apache.solr.ltr.feature.impl.FieldValueFeature",
+    "class": "org.apache.solr.ltr.feature.impl.FieldValueFeature",
```
Comment (mnilsson23) [inline solr/contrib/ltr/example/techproducts-features.json]: Can you grep for any other .json files which we need to change type=> class which we may have missed?

File: solr/contrib/ltr/example/solrconfig.xml [hunk 2]
```diff
@@ -876,11 +875,7 @@
        <str name="indent">true</str>
        <str name="df">text</str>
      </lst>
-<arr name="last-components">
-      <!-- Use the component in your requestHandler -->
-      <str>ltrComponent</str>
-    </arr>
-  </requestHandler>
+ </requestHandler>
```
Comment (mnilsson23) [inline solr/contrib/ltr/example/solrconfig.xml]: grep for other "searchComponent" sections in code or other configs and remove them too.  I think some tests do their own thing

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java [hunk 3]
```diff
@@ -44,10 +51,13 @@
...
  *
  */
-public class LTRQParserPlugin extends QParserPlugin {
+public class LTRQParserPlugin extends QParserPlugin implements ResourceLoaderAware, ManagedResourceObserver {
   public static final String NAME = "ltr";
 
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  
+  private ManagedFeatureStore fr = null;
+  private ManagedModelStore mr = null;
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java]: LTRQParser has its own ManagedModel/Feature store variables.  Should they be removed and use these new ones here?

File: solr/contrib/ltr/src/test/org/apache/solr/ltr/rest/TestModelManager.java [hunk 4]
```diff
@@ -20,7 +20,7 @@
 import org.apache.solr.common.util.NamedList;
 import org.apache.solr.core.SolrResourceLoader;
 import org.apache.solr.ltr.TestRerankBase;
-import org.apache.solr.ltr.ranking.LTRComponent;
+import org.apache.solr.ltr.ranking.LTRQParserPlugin;
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/test/org/apache/solr/ltr/rest/TestModelManager.java]: Verify the changes you added work in cloud mode with shards + replicas, and when restarting the instances they still pick up the existing deployment managed resources + Add some tests verifying that's the case.  Jon added a TestSolrCloud.java, which you can probably leverage to add more cases (if needed)

File: solr/contrib/ltr/example/techproducts-features.json (hunk 1 above)
Comment (diegoceccarelli) [inline solr/contrib/ltr/example/techproducts-features.json]: Done. We replaced all the occurrences

File: solr/contrib/ltr/example/solrconfig.xml (hunk 2 above)
Comment (diegoceccarelli) [inline solr/contrib/ltr/example/solrconfig.xml]: I can't find other occurrences

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java (hunk 3 above)
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java]: only the `ManagedModelStore`, now it is shared.

File: solr/contrib/ltr/example/solrconfig.xml (hunk 2 above)
Comment (mnilsson23) [inline solr/contrib/ltr/example/solrconfig.xml]: there's the example svm training python script as well which might have been missed

**Pull request merged**